lib/System/IO/TailFile.hs

```diff
@@ -38,27 +40,27 @@ import System.IO.Error (isDoesNotExistError)
 
     Data already existing in the file before `tailFile` is invoked is ignored.
  -}
-tailFile :: ByteString -- ^ System.Posix.ByteString.FilePath.RawFilePath 
+tailFile :: FilePath
          -> (a -> Data.ByteString.ByteString -> IO a) -- ^ State update function.
          -> IO a -- ^ Monadic action for getting the initial state.
          -> IO void -- ^ The result action never returns!
-tailFile filepath callback initial = withINotify (\i -> 
+tailFile filepath callback initial = withINotify (\i ->
     do state <- initial
        loop i state)
     where
     loop i =
         let go pristine a = do ea' <- tryJust (guard . isDoesNotExistError)
                                               (watchFile pristine i a)
-                               case ea' of 
+                               case ea' of
                                   Left ()  -> do threadDelay 5e5
                                                  go False a -- reuse the state
                                   Right a' -> go False a'
         in  go True
-    watchFile pristine i a = 
+    watchFile pristine i a =
         do sem <- newMVar mempty
-           bracket (addWatch i 
-                             [Modify,MoveSelf,DeleteSelf] 
-                             filepath 
+           bracket (addWatch i
+                             [Modify,MoveSelf,DeleteSelf]
+                             (Data.Text.Encoding.encodeUtf8 (Data.Text.pack filepath))
```

Comment:
This is a really tricky one - do we assume a UTF8 encoding? Afaik, glibc does, but Qt doesn't. It's really up to the application, the file system doesn't care...

Reply:
I had started the changes to support the latest hinotify, but wasn't sure how to handle RawFilePath. This looks good and the tests pass. Thanks!